metals/src/main/scala/scala/meta/internal/metals/debug/DebugServer.scala

```diff
@@ -105,6 +114,85 @@ object DebugServer {
     }
   }
 
+  def resolveMainClassParams(
+      params: DebugUnresolvedMainClassParams,
+      buildTargetClassesFinder: BuildTargetClassesFinder,
```

Comment:
Let's maybe move these new methods to `BuildTargetClassesFinder` with other than `params` method parameters added to the constructor? Also, with client instead of showWarningMessage.

Reply:
I am not sure. This method takes some debug parameters to resolve them to other parameters and reports about running debug on one class, if multiple found. So it's more about debug, then about finding, which is extracted to `Finder`

Reply:
Then maybe this should not actually be an object but maybe `DebugProvider`  class? Start also takes a lot of parameters, which are not specific to the actual function invocation.

Reply:
`DebugProvider` class created, all code from `DebugServer` object moved to it.